pkg/scheduler/objects/preemption_queue_test.go

```diff
@@ -53,101 +53,156 @@ type res struct {
-	parentQ.guaranteedResource = smallestRes
-	rootQ.allocatedResource = resources.Multiply(smallestRes, 2)
-	parentQ.allocatedResource = resources.Multiply(smallestRes, 2)
-	childQ2.allocatedResource = resources.Multiply(smallestRes, 2)
-	rootPreemptable, pPreemptable, cPreemptable1, cPreemptable2 = getPreemptableResource(rootQ, parentQ, childQ1, childQ2)
-	assert.Assert(t, resources.Equals(rootPreemptable, resources.Multiply(smallestRes, 2)), "usage is equal to guaranteed in root queue. so nothing to preempt")
-	assert.Assert(t, resources.Equals(pPreemptable, smallestRes), "usage has exceeded twice than guaranteed in parent queue. preemtable resource should be equal to guaranteed res")
-	assert.Assert(t, resources.IsZero(cPreemptable1), "nothing to preempt as no usage in child1 queue")
-	assert.Assert(t, resources.Equals(cPreemptable2, smallestRes), "usage has exceeded twice than guaranteed in child2 queue. preemtable resource should be equal to guaranteed res")
+	assertMessage1 := "nothing to preempt as no guaranteed set and no usage"
+	assertMessage2 := "usage has exceeded guaranteed, preemptable resource should be as expected"
+	assertMessage3 := "usage has extra resource type not defined as part of guaranteed, that extra should be preemptable"
+	assertMessage4 := "usage has exceeded guaranteed for some queues, preemptable resources should be as expected"
 
-	rootQ.guaranteedResource = resources.Multiply(smallestRes, 2)
-	parentQ.guaranteedResource = smallestRes
-	childQ2.guaranteedResource = smallestRes
-	rootQ.allocatedResource = nil
-	parentQ.allocatedResource = nil
-	childQ2.allocatedResource = nil
-	childRes := resources.NewResourceFromMap(map[string]resources.Quantity{"second": 5})
-	childQ1.guaranteedResource = childRes
-	rootPreemptable, pPreemptable, cPreemptable1, cPreemptable2 = getPreemptableResource(rootQ, parentQ, childQ1, childQ2)
-	assert.Assert(t, rootPreemptable.IsEmpty(), "nothing to preempt as no usage")
-	assert.Assert(t, pPreemptable.IsEmpty(), "nothing to preempt as no usage")
-	assert.Assert(t, cPreemptable1.IsEmpty(), "nothing to preempt as no usage")
-	assert.Assert(t, cPreemptable2.IsEmpty(), "nothing to preempt as no usage")
+	assertMessageStruct1 := assertMessage{assertMessage1, assertMessage1, assertMessage1, assertMessage1}
+	assertMessageStruct2 := assertMessage{assertMessage2, assertMessage2, assertMessage1, assertMessage2}
+	assertMessageStruct3 := assertMessage{assertMessage3, assertMessage3, assertMessage1, assertMessage1}
+	assertMessageStruct4 := assertMessage{assertMessage3, assertMessage3, assertMessage4, assertMessage1}
+
+	var tests = []struct {
+		testName  string
+		resources res
+	}{
+		{
+			// no guaranteed and no usage. so nothing to preempt
+			testName: "NoGuaranteedNoUsage",
+			resources: res{
+				guaranteed:     resource{nil, nil, nil, nil},
+				allocated:      allocated1,
+				remaining:      preemptable1,
+				assertMessages: assertMessageStruct1,
```

Comment:
This is ugly. It's difficult to see what is being passed in. Even though there's a bit of code duplication here, setting the values explicitly here makes the tests clearer.

Reply:
Make sense, Done.